pkg/controller/pod.go

```diff
@@ -835,6 +835,11 @@ func (c *Controller) reconcileRouteSubnets(cachedPod, pod *v1.Pod, needRoutePodN
 		podIP = pod.Annotations[fmt.Sprintf(util.IPAddressAnnotationTemplate, podNet.ProviderName)]
 		subnet = podNet.Subnet
 
+		// Check if pod uses nodeSwitch subnet
+		if subnet.Name == c.config.NodeSwitch {
+			return fmt.Errorf("NodeSwitch subnet %s is unavailable for pod", subnet.Name)
+		}
+
```

Comment:
can webhook limit this too？

Reply:
I find it is difficult to add webhook for pod use join subnet.  The name of the "join" subnet is implemented by setting the "node-switch" in ovn-controller. We can not get "node-switch" name in webhook because it dosen't have a controller.

Reply:
ok